src/github.ts

```diff
@@ -58,23 +58,20 @@ export const upsertComment = async ({
   })
 
   // find previous comments made by this action
-  existingComments.filter(item => item.body?.startsWith(hiddenHeader))
+  const coverageComments = existingComments.filter(item =>
+    item.body?.startsWith(hiddenHeader)
+  )
 
   // remove the last one from the list, this is the one we will update
-  const lastComment = existingComments.pop()
+  const lastComment = coverageComments.pop()
 
   // delete all existingComments (only need to keep one, the last one)
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const deletePromises: any[] = []
-
-  existingComments.map(async item =>
-    deletePromises.push(
-      github.issues.deleteComment({
-        owner: repoOwner,
-        repo: repoRepo,
-        comment_id: item.id
-      })
-    )
+  const deletePromises = coverageComments.map(async item =>
```

Comment:
This change is minor but still helpful. `map` returns an array of promises, exactly the same was collected to the `deletePromises` with `push`. All it brings proper typing and removes unwanted `any`.